storage/tokudb/tokudb_thread.h

```diff
@@ -32,97 +32,69 @@ Copyright (c) 2006, 2015, Percona and/or its affiliates. All rights reserved.
-    static const int sleep_step = 1000000;                                     \
-                                                                               \
-    remaining = *abs_timeout;                                                  \
-    while ((pthread_rc = lock_func_name(mutex)) == EBUSY) {                    \
-        ts.tv_sec = 0;                                                         \
-        ts.tv_nsec = (remaining.tv_sec > 0 ?                                   \
-                      sleep_step :                                             \
-                      _x_min(remaining.tv_nsec,sleep_step));                   \
-        nanosleep(&ts, &slept);                                                \
-        ts.tv_nsec -= slept.tv_nsec;                                           \
-        if (ts.tv_nsec <= remaining.tv_nsec) {                                 \
-            remaining.tv_nsec -= ts.tv_nsec;                                   \
-        } else {                                                               \
-            remaining.tv_sec--;                                                \
-            remaining.tv_nsec =                                                \
-              (sleep_step - (ts.tv_nsec - remaining.tv_nsec));                 \
-        }                                                                      \
-        if (remaining.tv_sec < 0 ||                                            \
-            (!remaining.tv_sec && remaining.tv_nsec <= 0)) {                   \
-            return ETIMEDOUT;                                                  \
-        }                                                                      \
-    }                                                                          \
-                                                                               \
-    return pthread_rc;                                                         \
-}
-
-timed_lock_define(pthread_mutex_timedlock,
-                  pthread_mutex_t,
-                  pthread_mutex_trylock);
-
-timed_lock_define(pthread_rwlock_timedrdlock,
-                  pthread_rwlock_t,
-                  pthread_rwlock_tryrdlock);
-
-timed_lock_define(pthread_rwlock_timedwrlock,
-                  pthread_rwlock_t,
-                  pthread_rwlock_trywrlock);
-
-#endif //(defined(__MACH__) || defined(__APPLE__)) && _POSIX_TIMERS <= 0
-
-uint my_tid(void);
-
-// Your basic mutex
-class mutex_t {
-public:
-    mutex_t(void);
-    ~mutex_t(void);
-
-    void lock(void);
-    int lock(ulonglong microseconds);
-    void unlock(void);
+    namespace thread {
+
+        uint my_tid(void);
+
+        // Your basic mutex
+        class mutex_t {
+           public:
+            mutex_t(pfs_key_t key);
+            mutex_t(void) : mutex_t(PFS_NOT_INSTRUMENTED) {}
+            ~mutex_t(void);
+
+            void reinit(pfs_key_t key);
```

Comment:
Either mutex_t::reinit, either rwlock_t::reinit is unused (not sure if the second one is required either, see the comment elsewhere)